### ISSUE TITLE
Enhancement: [naming-convention] Allow enforcing a naming convention for namespace imports

### ISSUE THREAD

**Issue by binoche9** [typescript-eslint/typescript-eslint#7164]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/naming-convention/

### Description

I propose that we enhance the naming-convention rule so that we can enforce naming conventions on namespace imports.

In my case I would like to standardize on PascalCase for namespace imports, but obviously other people are welcome to choose something else.

### Fail

```typescript
// Assuming I want PascalCase
import * as foo_bar_baz from "path/to/foo_bar_baz"
```

### Pass

```typescript
// Assuming I want PascalCase
import * as FooBarBaz from "path/to/foo_bar_baz"
```

### Additional Info

In my `Pass` example I have the namespace import match the file name since that's what I prefer, but I am not proposing that we build that enforcement into this rule - I'm only just proposing that the namespace import is in PascalCase

**Comment by bradzacher** [comment]
Duplicate of #2106
